Source/WebCore/page/NavigationHistoryEntry.cpp

```diff
@@ -142,4 +146,12 @@ void NavigationHistoryEntry::setState(RefPtr<SerializedScriptValue>&& state)
     m_associatedHistoryItem->setNavigationAPIStateObject(WTFMove(state));
 }
 
+auto NavigationHistoryEntry::DocumentState::fromContext(ScriptExecutionContext* context) -> DocumentState
+{
+    return {
+        context ? std::optional { context->identifier() } : std::nullopt,
+        context ? context->referrerPolicy() : ReferrerPolicy::Default
+    };
```

Comment:
Have you considered this (it is one less check)?

if (context)
    return { std::optional { context->identifier() }, context->referrerPolicy() };
return { std::nullopt, ReferrerPolicy::Default };